### DISCUSSION TITLE
Is there documentation on the Gruntwork pipeline tool (Concept + Architecture incl. supported branching model and code example) for Terragrunt?

### DISCUSSION THREAD

**Discussion by marijakstrazdas** [gruntwork-io/knowledge-base#149] [Help] [answered]
Is there documentation on the Gruntwork pipeline tool (Concept + Architecture incl. supported branching model and code example) for Terragrunt?

**Comment by zackproser** [comment, marked as answer]
Our current best guide for this is here: https://docs.gruntwork.io/docs/guides/build-it-yourself/pipelines/

Specifically, the Concept and Architecture discussion begins here: https://docs.gruntwork.io/docs/guides/build-it-yourself/pipelines/core-concepts/why-is-it-important-to-have-ci-cd/

The code samples are available in the section that details how to configure Pipelines on your own: https://docs.gruntwork.io/docs/guides/build-it-yourself/pipelines/deployment-walkthrough/pre-requisites

We are aware that these docs are currently out of date, but the core information (Architecture, Risk modeling, how to deploy it yourself) are available in this guide.

Luckily, if you purchase a Reference Architecture from us, you'll get Pipelines configured and working out of the box.
